common/djangoapps/entitlements/api/v1/views.py

```diff
@@ -274,30 +274,43 @@ def destroy(self, request, uuid):
             )
 
         if is_refund and entitlement.is_entitlement_refundable():
-            with transaction.atomic():
-                # Revoke and refund the entitlement
-                if entitlement.enrollment_course_run is not None:
-                    self._unenroll_entitlement(
-                        entitlement=entitlement,
-                        course_run_key=entitlement.enrollment_course_run.course_id,
-                        user=request.user
+            # Revoke the Course Entitlement and issue Refund
+            log.info(
+                'Entitlement Refund requested for Course Entitlement[%s]',
+                str(entitlement.uuid)
+            )
+            refund_status = refund_entitlement(course_entitlement=entitlement)
+
+            if refund_status:
+                with transaction.atomic():
```

Comment:
We might want to consider wrapping this in try/catch so that we can log if something goes wrong.

Reply:
Since the logic contained within this transaction is essentially the same as the revoke logic that we run in https://github.com/edx/edx-platform/blob/c25a20c5de6ddf84c775de8df25656769c10c97b/common/djangoapps/entitlements/api/v1/views.py#L86, we might want to consider moving it to a utility method that could be used in both places. Doing so would help us keep the logic consistent between the two places and make it easier to fix bugs.